emscripten.py

```diff
@@ -1536,16 +1548,20 @@ def create_receiving(function_table_data, function_tables_defs, exported_impleme
 
   if not shared.Settings.SWAPPABLE_ASM_MODULE:
     if shared.Settings.DECLARE_ASM_MODULE_EXPORTS:
-      receiving += ';\n'.join(['var ' + s + ' = Module["' + s + '"] = asm["' + s + '"]' for s in module_exports])
-      # TODO: Instead of the above line, we would like to use the two lines below for smaller size version of exports; but currently JS optimizer is wired to look for the exact above syntax when analyzing
-      # exports.
-#      receiving += ''.join(['var ' + s + ' = asm["' + s + '"];\n' for s in module_exports])
-#      receiving += 'for(var module_exported_function in asm) Module[module_exported_function] = asm[module_exported_function];\n'
+      if shared.Settings.WASM and shared.Settings.MINIMAL_RUNTIME:
+        receiving += '\n'.join([s + ' = asm["' + s + '"];' for s in module_exports]) + '\n'
+      else:
+        if shared.Settings.MINIMAL_RUNTIME:
+          receiving += '\n'.join(['var ' + s + ' = asm["' + s + '"];' for s in module_exports]) + '\n'
```

Comment:
See this comment: https://github.com/emscripten-core/emscripten/pull/7923#discussion_r250689674

Because in wasm compilation is asynchronous, it is not possible to process the exports at the top level of the script. I.e. when in asm.js one can do

```js
var asm = Module["asm"](asmGlobalArg, Module.asmLibraryArg, buffer);
var _main = asm["_main"];
```

in wasm one must do

```js
var _main;
WebAssembly.instantiate(Module["wasm"], imports).then((function(output) {
 var asm = output.instance.exports;
 _main = asm["P"];
 ...
}));

because 

```js
WebAssembly.instantiate(Module["wasm"], imports).then((function(output) {
 var asm = output.instance.exports;
 var _main = asm["P"];
 ...
}));

would create the function in wrong scope.